src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2231,6 +2237,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         ASR::Tuple_t* tuple_type_right = ASR::down_cast<ASR::Tuple_t>(ASRUtils::expr_type(x.m_right));
         std::string type_code_right = ASRUtils::get_type_code(tuple_type_right->m_type,
                                                         tuple_type_right->n_type);
+        ASR::Tuple_t* tuple_type = new ASR::Tuple_t();
```

Comment:
Don't initialize ASR types like this --- it leaks memory and is slow. Use the `ASR::make_Tuple_t` function.